server-x/tools/add-keycloak.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash -e
+
+###########################
+# Download Keycloak #
+###########################
+
+echo "Keycloak from [download]: $KEYCLOAK_DIST"
+
+cd /opt/jboss/
+curl -L $KEYCLOAK_DIST | tar zx
```

Comment:
Honestly, I've never likes this approach with building the image from a bash script. One of the biggest drawbacks is that you need to perform a "clean" build (clean up whatever temporary files you created) and it also requires you to install additional libraries, such as "tar".

A much better approach is to use the "builder images". You might have a look how we did this in the Operator: https://github.com/keycloak/keycloak-operator/blob/master/Dockerfile

Reply:
Humm ... I read about multi-stage builds but I did not figure out how I could use it here. Your example seems to help.

So, we would just use a step to both download and extract the distribution from the URL?